### ISSUE TITLE
Bug with asset's base64 encoded strings containing new lines causing 'URL(string: dataString)' to return nil

### ISSUE THREAD

**Issue by mqzkim** [airbnb/lottie-ios#2010]
There is a bug where the 'URL(string: dataString)' function returns nil when initializing Data with a base64 encoded string from the assets area that contains new lines, even though it works fine on Android and Web platforms with the same string. While it's not typical for assets to contain new lines, it can happen during the optimization process of Lottie files and could cause issues globally. For a workaround, I suggests using the 'trimmingCharacters' function of String to remove the new lines.
https://github.com/airbnb/lottie-ios/blob/cc7b5c67189311fdaede48fcb18588d1ac1b4ff1/Sources/Private/Model/Assets/ImageAsset.swift#L85

```
  internal init?(dataString: String, options: DataURLReadOptions = []) {
      let trimmedDataString = dataString.trimmingCharacters(in: .whitespacesAndNewlines)
    guard
      dataString.hasPrefix("data:"),
      let url = URL(string: trimmedDataString)
      else {
      return nil
    }
      
    // The code below is needed because Data(contentsOf:) floods logs
    // with messages since url doesn't have a host. This only fixes flooding logs
    // when data inside Data URL is base64 encoded.
    if
      let base64Range = trimmedDataString.range(of: ";base64,"),
      !options.contains(DataURLReadOptions.legacy)
    {
      let encodedString = String(trimmedDataString[base64Range.upperBound...])
      self.init(base64Encoded: encodedString)
    } else {
      try? self.init(contentsOf: url)
    }
  }
```

## Which Version of Lottie are you using?

Lottie 4.1.2

## Expected Behavior 

https://user-images.githubusercontent.com/10410095/228734285-f1198fd7-f456-40f1-a9a7-e9fa2224541b.mp4

## Actual Behavior

https://user-images.githubusercontent.com/10410095/228734289-732f68fd-7d76-45f4-b173-9a084bf0cd5a.mp4

## Animation JSON

[spot-error.json.zip](https://github.com/airbnb/lottie-ios/files/11107917/spot-error.json.zip)

**Comment by calda** [comment]
Thanks for the report @mqzkim! Since you already have a solution in mind, would you be able to post a pull request with the fix?

**Comment by rohit9934** [comment]
I also faced this issue, good solution @mqzkim , please pull a PR or shall i do it?

**Comment by calda** [comment]
Fixed in #2016
